### ISSUE TITLE
Error when building with electron-builder

### ISSUE THREAD

**Issue by matthpmore** [th-ch/youtube-music#246]
When I run `yarn build` it outputs the following:
```
$ yarn run clean && electron-builder --win --mac --linux
$ rimraf dist
  • electron-builder  version=22.10.5 os=10.0.19042
  • loaded configuration  file=package.json ("build" field)
  • writing effective config  file=dist\builder-effective-config.yaml
  ⨯ Build for macOS is supported only on macOS, please see https://electron.build/multi-platform-build
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
I figured the --mac and --linux flags where causing this error, so I decided to run `yarn run clean && electron-builder --win` but it still gives me an error:
```
yarn run v1.22.10
$ rimraf dist
Done in 0.39s.
  • electron-builder  version=22.10.5 os=10.0.19042
  • loaded configuration  file=package.json ("build" field)
  • writing effective config  file=dist\builder-effective-config.yaml
  • rebuilding native dependencies  dependencies=registry-js@1.15.0 platform=win32 arch=x64
  • install prebuilt binary  name=registry-js version=1.15.0 platform=win32 arch=x64
  • build native dependency from sources  name=registry-js
                                          version=1.15.0
                                          platform=win32
                                          arch=x64
                                          reason=prebuild-install failed with error (run with env DEBUG=electron-builder to get more information)
                                          error=prebuild-install info begin Prebuild-install version 6.1.2
    prebuild-install WARN install prebuilt binaries enforced with --force!
    prebuild-install WARN install prebuilt binaries may be out of date!
    prebuild-install info looking for local prebuild @ prebuilds\registry-js-v1.15.0-electron-v85-win32-x64.tar.gz
    prebuild-install info looking for cached prebuild @ C:\Users\mathe\AppData\Roaming\npm-cache\_prebuilds\65be2c-registry-js-v1.15.0-electron-v85-win32-x64.tar.gz
    prebuild-install http request GET https://github.com/desktop/registry-js/releases/download/v1.15.0/registry-js-v1.15.0-electron-v85-win32-x64.tar.gz
    prebuild-install http 404 https://github.com/desktop/registry-js/releases/download/v1.15.0/registry-js-v1.15.0-electron-v85-win32-x64.tar.gz
    prebuild-install WARN install No prebuilt binaries found (target=11.4.4 runtime=electron arch=x64 libc= platform=win32)

  ⨯ cannot execute  cause=exit status 1
                    out=
    > registry-js@1.15.0 install C:\Users\mathe\Desktop\youtube-music-master\node_modules\registry-js
    > prebuild-install || node-gyp rebuild


    C:\Users\mathe\Desktop\youtube-music-master\node_modules\registry-js>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
    Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
      main.cc
      win_delay_load_hook.cc
    C:\Users\mathe\Desktop\youtube-music-master\node_modules\node-addon-api\napi-inl.h(1575,24): error C3861: 'napi_is_detached_arraybuffer': identifier not found (compiling source file ..\src\main.cc) [C:\Users\mathe\Desktop\youtube-music-master\node_modules\registry-js\build\registry.vcxproj]
    C:\Users\mathe\Desktop\youtube-music-master\node_modules\node-addon-api\napi-inl.h(1581,24): error C3861: 'napi_detach_arraybuffer': identifier not found (compiling source file ..\src\main.cc) [C:\Users\mathe\Desktop\youtube-music-master\node_modules\registry-js\build\registry.vcxproj]

                    errorOut=gyp ERR! build error
    gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\MSBuild.exe` failed with exit code: 1
    gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:194:23)
    gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
    gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
    gyp ERR! System Windows_NT 10.0.19042
    gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
    gyp ERR! cwd C:\Users\mathe\Desktop\youtube-music-master\node_modules\registry-js
    gyp ERR! node -v v14.16.1
    gyp ERR! node-gyp -v v5.1.0
    gyp ERR! not ok
    npm ERR! code ELIFECYCLE
    npm ERR! errno 1
    npm ERR! registry-js@1.15.0 install: `prebuild-install || node-gyp rebuild`
    npm ERR! Exit status 1
    npm ERR!
    npm ERR! Failed at the registry-js@1.15.0 install script.
    npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

    npm ERR! A complete log of this run can be found in:
    npm ERR!     C:\Users\mathe\AppData\Roaming\npm-cache\_logs\2021-05-02T04_31_11_980Z-debug.log

                    command=npm.cmd rebuild registry-js@1.15.0
                    workingDir=
```
I can't figure out what's wrong. I've tried deleting the node_modules folder and installing everything again, but it doesn't work.

**Comment by th-ch** [comment]
Hey @matthpmore, thanks for the feedback! I guess you are on Windows from the output above, can you try switching to [node 12](https://github.com/th-ch/youtube-music/blob/master/.github/workflows/build.yml#L21) (it seems you are using node 14) and running the following after having deleted the `node_modules` folder?
```
yarn cache clean
yarn --frozen-lockfile
yarn build:win
```
(it clears the yarn cache before re-installing dependencies)

Windows is built in [CI](https://github.com/th-ch/youtube-music/blob/master/.github/workflows/build.yml#L57) and [last builds look ok](https://github.com/th-ch/youtube-music/runs/2470850192) so it should work fine but issues linked to different setups are always possible 😬

**Comment by matthpmore** [comment]
It worked perfectly, thank you so much! The issue was definitely the node version. I don't know much about programming so I didn't know I had to use version 12. By the way, I tried version 12.9 at first because I didn't notice there were more recent ones, but after it didn't work I found out there was version 12.22 which is the one I used. Just a heads up. Sorry for the inconvenience, thank you again.
